src/pages/SupplierRequisitionsPage.js

```diff
@@ -239,7 +239,7 @@ export class SupplierRequisitionsPage extends React.Component {
           queryString="name BEGINSWITH[c] $0"
           sortByString="name"
           onSelect={name => this.onNewRequisition({ otherStoreName: name })}
-          onClose={this.onNewRequisition}
+          onClose={() => this.onNewRequisition()}
```

Comment:
This seems functionally equivalent?

Reply:
as above:

Event object was being passed, which caused the check for a parameter being passed return true, creating the requisition.

This results in no parameter being passed, and the check to then evaluate to false, closing the modal

Reply:
Ah right. I think I raised somewhere (maybe I just thought it) when looking at this in the past that passing the same method `onNewRequisition` into onClose wasn't well separated, that is to say another method would be cleaner:
```javascript
onCancelNewRequisition = () => this.setState({ byProgramModalOpen: false, isCreatingRequisition: false });
```
Seeing as I just looked at it,  I think `byProgramModalOpen`, `isCreatingRequisition` and `usesPrograms` could be used a bit differently less complexly.